src/pages/TransactionsList/TransactionsList.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useContext, useMemo, useRef, useState } from 'react';
+import { Link } from 'react-router-dom';
+import Filter from '../../components/Filter/Filter';
+import ItemsList from '../../components/ItemsList/ItemsList';
+import DataContext from '../../contexts/DataContext';
+import { FiltersContext } from '../../contexts/FiltersContext';
+import { FilterName, FilterType } from '../../interfaces/Filter';
+import Transaction from './Transaction/Transaction';
+import styles from './TransactionsList.module.css';
+
+const TransactionsList = () => {
+	const { transactions, transactionFilters } = useContext(DataContext);
+	const { appliedTransactionFilters, setAppliedTransactionFilters } = useContext(FiltersContext);
+	const debounceTimeoutId = useRef<NodeJS.Timeout>(null);
+	const [page, setPage] = useState(1);
+
+	// TODO: remove filter logic from component
+	const filteredTransactions = useMemo(() => (transactions.filter((transaction) => {
+		return Object.entries(appliedTransactionFilters).every(([name, value]) => {
+			if (name === FilterName.CARD_ACCOUNT) {
+				return transaction.cardAccount === value;
+			}
+
+			if (name === FilterName.CARD_ID || name === FilterName.CURRENCY) {
+				return transaction[name] === value;
+			}
+
+			if (name === FilterName.AMOUNT) {
+				const filterValue = value as { min: number; max: number };
+				return transaction.amount >= filterValue.min && transaction.amount <= filterValue.max;
+			}
+
+			if (name === FilterName.DATE) {
+				const formatedDate: Date = new Date(value as string);
+				return transaction.transactionDate === `${formatedDate.getDate()}/${formatedDate.getMonth() + 1}/${formatedDate.getFullYear()}`;
+			}
+
+			return true;
+		}
+	)})), [transactions, appliedTransactionFilters]);
+		
+	const updateFilters = (value: string | { min: number, max: number }, name: FilterName) => {
```

Comment:
Logic duplication with `CardsList`, should be in a separate file, and imported in theses components